### DISCUSSION TITLE
How to include an optional image

### DISCUSSION THREAD

**Discussion by royfrancis** [typst/typst#3075] [Q&A]
In my **show.typ**, I have an image

```
 $if(bg-image)$
    bg-image: (
      path: "$bg-image.path$"
    ), 
  $endif$
```

In my **template.typ**, I use it like this:

`background: place(center + top, image(bg-image.path, height: 100%))`

How do I set this image to be optional? If no image is passed it crashes. I tried `[$bg-image.path$]` and `["$bg-image.path$"]` but that didn't work.

**Comment by istudyatuni** [comment]
How to pass something optional:

```typ
#let opt-image(path: none) = {
  if path != none {
    image(path)
  }
}
#opt-image()
#opt-image(path: "image.png")
```

**Reply by laurmaedje** [reply]
Small correction: The second call needs the name of the parameter.

**Comment by royfrancis** [comment]
```
background: if bg-image != none {
  place(center + top, image(bg-image.path, height: 100%))
}
```

**Reply by istudyatuni** [reply]
Could you share more code? It's hard to help without more context :(
